bgp/gobgp/plugin_impl_gobgp_test.go

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2017 Pantheon technologies s.r.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//Package gobgp_test contains Ligato GoBGP Plugin implementation tests
+package gobgp_test
+
+import (
+	"errors"
+	"github.com/ligato/bgp-agent/bgp"
+	"github.com/ligato/bgp-agent/bgp/gobgp"
+	"github.com/ligato/cn-infra/core"
+	"github.com/ligato/cn-infra/flavors/local"
+	"github.com/ligato/cn-infra/logging/logroot"
+	. "github.com/onsi/gomega"
+	"github.com/osrg/gobgp/config"
+	bgpPacket "github.com/osrg/gobgp/packet/bgp"
+	"github.com/osrg/gobgp/server"
+	"github.com/osrg/gobgp/table"
+	"sync"
+	"testing"
+	"time"
+)
+
+const (
+	nextHop1                string = "10.0.0.1"
+	nextHop2                string = "10.0.0.3"
+	prefix1                 string = "10.0.0.0"
+	prefix2                 string = "10.0.0.2"
+	length                  uint8  = 24
+	expectedReceivedAs             = uint32(65000)
+	maxSessionEstablishment        = 2 * time.Minute
+	timeoutForNotReceiving         = 5 * time.Second
+)
+
+var flavor = &local.FlavorLocal{}
+
+// TestGoBGPPluginInfoPassing tests gobgp plugin for the ability of retrieving of ReachableIPRoutes using BGP protocol and passing this information to its registered watchers.
+// Test is also testing the ability of watch unregistering.
+func TestGoBGPPluginInfoPassing(t *testing.T) {
+	// creating help variables/initialization
+	RegisterTestingT(t)
+	channel := make(chan bgp.ReachableIPRoute, 10)
+	var lifecycleWG sync.WaitGroup
+
+	// create components
+	routeReflector, err := createAndStartRouteReflector(routeReflectorConf)
+	Expect(err).To(BeNil(), "Can't create or start route reflector")
+	goBGPPlugin := createGoBGPPlugin(serverConf)
+
+	// watch -> start lifecycle of gobgp plugin -> send path to Route reflector -> check receiving on other end
+	watchRegistration, registrationErr := goBGPPlugin.WatchIPRoutes("TestWatcher", bgp.ToChan(channel, logroot.StandardLogger()))
+	Expect(registrationErr).To(BeNil(), "Can't properly register to watch IP routes")
+	Expect(watchRegistration).NotTo(BeNil(), "WatchRegistration must be non-nil to be able to close registration later")
+	lifecycleCloseChannel := startPluginLifecycle(goBGPPlugin, assertCorrectLifecycleEnd(&lifecycleWG))
```

Comment:
Given.AgentPluginGoBGP

(extract all intermediate assertions away from test scenario to make it readable)